src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/CreateCitizenClaimCallbackHandler.java

```diff
@@ -0,0 +1,116 @@
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.Payment;
+import uk.gov.hmcts.cmc.domain.models.PaymentStatus;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.CREATE_LEGAL_REP_CLAIM;
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CITIZEN;
+import static uk.gov.hmcts.cmc.domain.utils.LocalDateTimeFactory.nowInLocalZone;
+
+@Service
+public class CreateCitizenClaimCallbackHandler extends CallbackHandler {
+
+    private static final List<CaseEvent> EVENTS = Collections.singletonList(CREATE_LEGAL_REP_CLAIM);
+    private static final List<Role> ROLES = Collections.singletonList(CITIZEN);
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final IssueDateCalculator issueDateCalculator;
+    private final ReferenceNumberRepository referenceNumberRepository;
+    private final ResponseDeadlineCalculator responseDeadlineCalculator;
+    private final CaseMapper caseMapper;
+    private final PaymentsService paymentsService;
+
+    @Autowired
+    public CreateCitizenClaimCallbackHandler(
+        CaseDetailsConverter caseDetailsConverter,
+        IssueDateCalculator issueDateCalculator,
+        ReferenceNumberRepository referenceNumberRepository,
+        ResponseDeadlineCalculator responseDeadlineCalculator,
+        CaseMapper caseMapper,
+        PaymentsService paymentsService
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.issueDateCalculator = issueDateCalculator;
+        this.referenceNumberRepository = referenceNumberRepository;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.caseMapper = caseMapper;
+        this.paymentsService = paymentsService;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(CallbackType.ABOUT_TO_SUBMIT, this::createCitizenClaim);
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private CallbackResponse createCitizenClaim(CallbackParams callbackParams) {
+        logger.info("Created citizen case for callback of type {}", callbackParams.getType());
+        Claim claim = caseDetailsConverter.extractClaim(callbackParams.getRequest().getCaseDetails());
+
+        LocalDate issuedOn = issueDateCalculator.calculateIssueDay(nowInLocalZone());
+        LocalDate responseDeadline = responseDeadlineCalculator.calculateResponseDeadline(issuedOn);
+        String referenceNumber = referenceNumberRepository.getReferenceNumberForLegal();
```

Comment:
`getReferenceNumberForLegal` is wrong here, please use the one we have for citizen

Reply:
good spot, thanks 👍